sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java

```diff
@@ -397,12 +396,14 @@ public WatermarkEstimatorStateT getState() {
   }
 
   public static class DefaultGetSize {
-    /** Uses {@link Sizes.HasSize} to produce the size. */
+    /** Uses {@link HasProgress} to produce the size. */
     @SuppressWarnings("unused")
     public static <InputT, OutputT> double invokeGetSize(
         DoFnInvoker.ArgumentProvider<InputT, OutputT> argumentProvider) {
-      if (argumentProvider.restrictionTracker() instanceof HasSize) {
-        return ((HasSize) argumentProvider.restrictionTracker()).getSize();
+      if (argumentProvider.restrictionTracker() instanceof HasProgress) {
```

Comment:
I'm not very familiar with how this invoker works but I thought we check whether `restrictionTracker ` has `HasSize`, if not then fallback to progress.

Reply:
DoFn can provide a method annotated with `@GetSize` and the fallback is the HasProgress on the RestrictionTracker. This is analogous to RestrictionProvider/RProvider being able to provide the size while the RestrictionTracker/RTracker provide progress in Python/Go.

Reply:
Thanks for explaining this! If so, do we still need to keep `GetSize`? It seems like highly overlap with `HasProgress`.

Reply:
Yes, because it allows the user to provide an override on how size is computed without needing to implement a restriction tracker. This is similar to how a restriction provider can compute the size.